### PR TITLE
Added hopsfs-s3 support

### PR DESCRIPTION
**Pull request by smkniazi**

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -35,9 +35,21 @@
 default['hops']['conf_dir']                    = node['hops']['conf_dir_parent'] + "/hadoop"
 default['hops']['share_dir']                    = node['hops']['base_dir'] + "/share/hadoop"
 
-default['hops']['dn']['data_dir']              = "file://" + node['hops']['data_dir'] + "/hdfs/dn"
-default['hops']['dn']['data_dir_permissions']  = '700'
-default['hops']['nn']['name_dir']              = "file://" + node['hops']['data_dir'] + "/hdfs/nn"
+default['hops']['enable_cloud_storage']        = "false"
+default['hops']['cloud_provider']              = "AWS"
```
Comment (kouzant) [inline attributes/default.rb]: Can you default this to `install/cloud` attribute? At least when we're doing a full AWS installation we won't need to set it in two places.

File: recipes/_config.rb [hunk 2]
```diff
@@ -13,3 +13,10 @@
     node.override['hops']['capacity']['resource_calculator_class'] = "org.apache.hadoop.yarn.util.resource.DominantResourceCalculatorGPU"    
   end
 end
+
+# Override install binaries for enterprise edition
+if node['install']['enterprise']['install'].casecmp? "true"
+  node.override['hops']['dist_url']    = "#{node['install']['enterprise']['download_url']}/hopshadoop/hops-#{node['hops']['version']}.tgz"
```
Comment (kouzant) [inline recipes/_config.rb]: If there is no reason, can you remove the `hopshadoop` directory to avoid confusion? It should be the mirror of CE

**Comment by kouzant** [comment]
What's the status of https://github.com/logicalclocks/hops-hadoop-chef/pull/170 ?

File: recipes/_config.rb (hunk 2 above)
Comment (smkniazi) [reply recipes/_config.rb]: limitation of archive.logicalclocks.com. Can not copy files to root dir

File: attributes/default.rb (hunk 1 above)
Comment (smkniazi) [reply attributes/default.rb]: fxd

**Pull request merged**